monai/inferers/inferer.py

```diff
@@ -226,14 +224,24 @@ def _initialize_mergers(self, inputs, outputs, patches, batch_size):
             out_patch = torch.chunk(out_patch_batch, batch_size)[0]
             # calculate the ratio of input and output patch sizes
             ratio = tuple(op / ip for ip, op in zip(in_patch.shape[2:], out_patch.shape[2:]))
-            ratios.append(ratio)
-            # calculate output_shape only if it is not provided and splitter is not None.
-            if self.splitter is not None and "output_shape" not in self.merger_kwargs:
-                output_shape = self._get_output_shape(inputs, out_patch, ratio)
-                merger = self.merger_cls(output_shape=output_shape, **self.merger_kwargs)
-            else:
-                merger = self.merger_cls(**self.merger_kwargs)
+
+            # calculate merging_shape and final_shape
+            merger_kwargs = self.merger_kwargs.copy()
+            final_shape, merging_shape = self._get_merged_shapes(inputs, out_patch, ratio)
```

Comment:
if `splitter` is None then these two values will be None as well in which case they have to be provided in `merger_kwargs`. I think this relationship needs to be mentioned in the docstring somewhere, what is says is that `splitter` can be None when assuming inputs are split into patches but it's not clear what the implications of that are. Perhaps this should be checked in the constructor?